collector/lib/CollectorConfig.cpp

```diff
@@ -398,6 +399,9 @@ void CollectorConfig::HandleSinspEnvVars() {
 }
 
 bool CollectorConfig::YamlConfigToConfig(YAML::Node& yamlConfig) {
+  // Don't read the file during a scrape
+  std::lock_guard<std::mutex> lock(*mutex_);
```

Comment:
With the changes to the type of mutex being used, this needs to be:
```suggestion
  std::unique_lock<std::shared_mutex> lock(mutex_);
```